lib/ViewModels/WorkbenchItemControls.ts

```diff
@@ -0,0 +1,27 @@
+import remove from "lodash-es/remove";
+import { action } from "mobx";
+import { BaseModel } from "../Models/Definition/Model";
+import { SelectableDimension } from "../Models/SelectableDimensions/SelectableDimensions";
+import ViewState from "../ReactViewModels/ViewState";
+
+type WorkbenchItemControlsGenerator = (
+  item: BaseModel
+) => SelectableDimension | undefined;
```

Comment:
Could we extend `SelectableDimension` type here to make it more specific to what we want people to provide in the return of their generator function? See other comments for this file. This could help us to provide specific intellisense hints for what people should be providing for this object: 
```
{
          type: "button",
          value: "Test Button",
          setDimensionValue: () => console.log('xxxxxxxx'),
          buttonStyle: "primary"
        }
        ```

Actually this is probably not good practise, but givent he complex inheritance in `SelectableDimensions.ts` we need a guide for how users of the plugin API can use `addControl` . How could we make this clear for people given the wide variety of controls that they could add?